### ISSUE TITLE
Support other view frameworks other than JSP

### ISSUE THREAD

**Issue by jwest84** [StripesFramework/stripes#36]
It would be nice if by bringing in Stripes, we didn't have to bring in all the JSP tags with it.  Could we possibly have Stripes-framework and then separate Stripes JSP.

**Comment by vankeisb** [comment]
Is it causing a problem ? Or the intent is to save a few Kbs in the stripes jar ?

**Comment by jwest84** [comment]
No problem. It's just an idea to make it possible for the open source community to build integrations with other front-end libraries.

**Comment by vankeisb** [comment]
Ok. So you mean factor out the "core" tagblib functionality into reusable java code ? 
I'm asking because Stripes is not tied to a view technology. You can use freemarker or anything else for that matter. It is pluggable. It's only the tags that are hard to reuse outside of jsp...

**Comment by jwest84** [comment]
Yeah, I have used Velocity with Stripes. My biggest issue is that it's hard to use some of the Stripes functionality because the logic is hardcoded into the JSP tags.  So you are correct, bringing the taglib stuff into reusable Java is really what I am looking for.

**Comment by vankeisb** [comment]
I could not agree more. 

I've had the same problem with TTT (a Statically Typed Templating lib) when I integrated with Stripes.

I ended up writing a subset of a JSP engine in order to handle this. In the end, it's a bit scary, but it works.  It imposes some design though...

Have a look at this if you're interested :
https://github.com/pojosontheweb/ttt/tree/master/stripes#tag-library

The idea is to use the `try-resource` pattern in order to have blocks that mimic the nesting of tags.

This allowed me to reuse Stripes tags directly. Otherwise, it's quite a lot of work to rewrite the whole thing to make it reusable. Was clearly not intended to.

It also has the benefit of allowing to reuse virtually any taglib, not only Stripes'...
